tests/functionnal/KnowbaseItem.php

```diff
@@ -445,7 +445,7 @@ public function testCreateWithCategories()
         $kbitems_id2 = $kbitem->add([
             'name' => __FUNCTION__ . '_2',
             'answer' => __FUNCTION__ . '_2',
-            'knowbaseitemcategories_id' => [$kb_cat_id1, $kb_cat_id2],
```

Comment:
Why changing the `knowbaseitemcategories_id` input key ? It may break existing plugins and API based scripts that are using this key.
For instance, `datainjection` seems to use it.

Reply:
In my tests the original input was not working and only kept a single value even if multiples are supplied.

It seems to me that `knowbaseitemcategories_id` was a foreign key field in the `glpi_knowbaseitems` table back when the relationship was 1-1.
The field was removed from the table as the relation became N-N using the `glpi_knowbaseitems_knowbaseitemcategories` link table.

I doesn't seem unimaginable to me that such a database change in the core impact the APIs / plugins.

It also seems wrong to store multiples value in a deleted fk field just for backward compatibility.
A proper fields that indicate that it isn't part of the table (`_` prefix) and have a clear name that indicate support for multiples values seems better to me.

Maybe I could add a extra case in the code to check for `knowbaseitemcategories_id` and convert it to  `_categories` with a depreciation warning ?

Reply:
Done [0854ebf](https://github.com/glpi-project/glpi/pull/11892/commits/0854ebf887fb21dcbc3aa1c5f1ef85925131db19).